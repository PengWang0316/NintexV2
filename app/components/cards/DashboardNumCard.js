import React from 'react';
import PropTypes from 'prop-types';
import { Box, Card, CardContent } from '@material-ui/core';
import { withStyles } from '@material-ui/core/styles';

const styles = {
  card: {
    width: 255,
    margin: '0 10px',
  },
  cardDiv: {
    display: 'flex',
    justifyContent: 'center',
    marginTop: 20,
  },
  contentDiv: {
    display: 'flex',
    justifyContent: 'space-between',
  },
};
// The regular experssion to help format numbers to have a comma in every three digitals
const NUM_FORMATE_REGEXP = /(\d)(?=(\d{3})+(?!\d))/g;

export const DashboardNumCard = ({
  classes, displayNum, title, icon, extraContent,
}) => (
  <Card className={classes.card}>
    <CardContent>
      <div className={classes.contentDiv}>
        <div>
          <Box fontSize={14} color="textSecondary">
            {title}
          </Box>
          <Box fontSize={26} fontWeight={1600} color="textSecondary">
            {displayNum ? displayNum.toString().replace(NUM_FORMATE_REGEXP, '$1,') : '--'}
          </Box>
        </div>
        {icon}
      </div>
      {extraContent}
    </CardContent>
  </Card>
);
DashboardNumCard.propTypes = {
  classes: PropTypes.objectOf(PropTypes.string).isRequired,
  displayNum: PropTypes.number,
  title: PropTypes.string.isRequired,
  icon: PropTypes.objectOf(PropTypes.any).isRequired,
  extraContent: PropTypes.element,
};
DashboardNumCard.defaultProps = { displayNum: null, extraContent: null };
export default withStyles(styles)(DashboardNumCard);
