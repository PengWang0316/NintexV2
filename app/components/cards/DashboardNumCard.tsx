import React, { ReactElement, memo } from 'react';
import { Box, Card, CardContent } from '@material-ui/core';
import { makeStyles } from '@material-ui/core/styles';

interface Props {
  displayNum: string | number | null;
  title: string;
  icon: ReactElement;
  extraContent?: ReactElement | null;
}

const useStyles = makeStyles({
  card: {
    width: 255,
    margin: '0 10px',
  },
  cardDiv: {
    display: 'flex',
    justifyContent: 'center',
    marginTop: 20,
  },
  contentDiv: {
    display: 'flex',
    justifyContent: 'space-between',
  },
});
// The regular experssion to help format numbers to have a comma in every three digitals
const NUM_FORMATE_REGEXP = /(\d)(?=(\d{3})+(?!\d))/g;

export const DashboardNumCard = ({
  displayNum = null, title, icon, extraContent = null,
}: Props): ReactElement => {
  const classes = useStyles({});
  return (
    <Card className={classes.card}>
      <CardContent>
        <div className={classes.contentDiv}>
          <div>
            <Box fontSize={14} color="textSecondary">
              {title}
            </Box>
            <Box fontSize={26} fontWeight={1600} color="textSecondary">
              {displayNum ? displayNum.toString().replace(NUM_FORMATE_REGEXP, '$1,') : '--'}
            </Box>
          </div>
          {icon}
        </div>
        {extraContent}
      </CardContent>
    </Card>
  );
};
export default memo(DashboardNumCard);
