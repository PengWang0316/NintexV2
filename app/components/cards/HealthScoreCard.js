/* eslint-disable react/jsx-one-expression-per-line */
import React, { useEffect } from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import { connect } from 'react-redux';
import { lightBlue, red, lightGreen } from '@material-ui/core/colors';
import { SwapVerticalCircle } from '@material-ui/icons';
import { Box } from '@material-ui/core';
import I18n from '@kevinwang0316/i18n';

import DashboardNumCard from './DashboardNumCard';
import { fetchInstanceStatus as fetchInstanceStatusAction } from '../../actions/InstanceActions';

const styles = {
  cardIcon: {
    fontSize: 36,
    color: lightBlue[500],
  },
  flexDiv: { display: 'flex' },
  box: { marginRight: 10 },
};

export const HealthScoreCard = ({ classes, instanceStatus, fetchInstanceStatus }) => {
  useEffect(() => {
    if (!instanceStatus) fetchInstanceStatus();
  });

  return (
    <DashboardNumCard
      displayNum={instanceStatus ? instanceStatus.completedPercentage : null}
      title={I18n.get('cardTitleWorkflowHealth')}
      icon={<SwapVerticalCircle className={classes.cardIcon} />}
      extraContent={(
        <div className={classes.flexDiv}>
          <Box className={classes.box} fontSize={14} color={red[500]}>Failed: {instanceStatus ? instanceStatus.failedPercentage : '--'}</Box>
          <Box fontSize={14} color={lightGreen[500]}>Started: {instanceStatus ? instanceStatus.startedPercentage : '--'}</Box>
        </div>
      )}
    />
  );
};

HealthScoreCard.propTypes = {
  classes: PropTypes.objectOf(PropTypes.string).isRequired,
  instanceStatus: PropTypes.objectOf(PropTypes.oneOfType([PropTypes.number, PropTypes.string])),
  fetchInstanceStatus: PropTypes.func.isRequired,
};
HealthScoreCard.defaultProps = {
  instanceStatus: null,
};
const mapStateToProps = state => ({ instanceStatus: state.instanceStatus });
const mapDispatchToProps = { fetchInstanceStatus: fetchInstanceStatusAction };
export default connect(mapStateToProps, mapDispatchToProps)(withStyles(styles)(HealthScoreCard));
