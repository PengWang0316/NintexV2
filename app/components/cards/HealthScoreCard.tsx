/* eslint-disable react/jsx-one-expression-per-line */
import React, { useEffect, memo } from 'react';
import { makeStyles } from '@material-ui/core/styles';
import { connect } from 'react-redux';
import { lightBlue, red, lightGreen } from '@material-ui/core/colors';
import { SwapVerticalCircle } from '@material-ui/icons';
import { Box } from '@material-ui/core';
import I18n from '@kevinwang0316/i18n';

import DashboardNumCard from './DashboardNumCard';
import { fetchInstanceStatus as fetchInstanceStatusAction } from '../../store/InstanceStatus/actions';
import { AppState } from '../../store/ConfigureStore';

const useStyles = makeStyles({
  cardIcon: {
    fontSize: 36,
    color: lightBlue[500],
  },
  flexDiv: { display: 'flex' },
  box: { marginRight: 10 },
});
let isFetching = false;

interface Props {
  instanceStatus: {
    completedPercentage: string | number | null;
    failedPercentage: string | number;
    startedPercentage: string | number;
  };
  fetchInstanceStatus: () => void;
}

export const HealthScoreCard = ({ instanceStatus, fetchInstanceStatus }: Props) => {
  const classes = useStyles({});
  useEffect(() => {
    if (!instanceStatus && !isFetching) {
      fetchInstanceStatus();
      isFetching = true;
    }
  });

  return (
    <DashboardNumCard
      displayNum={instanceStatus ? instanceStatus.completedPercentage : null}
      title={I18n.get('cardTitleWorkflowHealth')}
      icon={<SwapVerticalCircle className={classes.cardIcon} />}
      extraContent={(
        <div className={classes.flexDiv}>
          <Box className={classes.box} fontSize={14} color={red[500]}>Failed: {instanceStatus ? instanceStatus.failedPercentage : '--'}</Box>
          <Box fontSize={14} color={lightGreen[500]}>Started: {instanceStatus ? instanceStatus.startedPercentage : '--'}</Box>
        </div>
      )}
    />
  );
};

const mapStateToProps = (state: AppState) => ({ instanceStatus: state.instanceStatus });
const mapDispatchToProps = { fetchInstanceStatus: fetchInstanceStatusAction };
export default connect(mapStateToProps, mapDispatchToProps)(memo(HealthScoreCard));
