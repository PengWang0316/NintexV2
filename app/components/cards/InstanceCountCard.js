import React, { useEffect } from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import { connect } from 'react-redux';
import { orange } from '@material-ui/core/colors';
import { GroupWork } from '@material-ui/icons';
import I18n from '@kevinwang0316/i18n';

import DashboardNumCard from './DashboardNumCard';
import { fetchInstanceCount as fetchInstanceCountAction } from '../../actions/InstanceActions';

const styles = {
  cardIcon: {
    fontSize: 36,
    color: orange[500],
  },
};

export const InstanceCountCard = ({ classes, instanceCount, fetchInstanceCount }) => {
  useEffect(() => {
    if (!instanceCount) fetchInstanceCount();
  });

  return <DashboardNumCard displayNum={instanceCount} title={I18n.get('cardTitleWorkflowInstances')} icon={<GroupWork className={classes.cardIcon} />} />;
};

InstanceCountCard.propTypes = {
  classes: PropTypes.objectOf(PropTypes.string).isRequired,
  instanceCount: PropTypes.number,
  fetchInstanceCount: PropTypes.func.isRequired,
};
InstanceCountCard.defaultProps = {
  instanceCount: null,
};
const mapStateToProps = state => ({ instanceCount: state.instanceCount });
const mapDispatchToProps = { fetchInstanceCount: fetchInstanceCountAction };
export default connect(mapStateToProps, mapDispatchToProps)(withStyles(styles)(InstanceCountCard));
