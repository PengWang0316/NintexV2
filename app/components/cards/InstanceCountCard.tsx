import React, { useEffect, memo, ReactElement } from 'react';
import { makeStyles } from '@material-ui/core/styles';
import { connect } from 'react-redux';
import { orange } from '@material-ui/core/colors';
import { GroupWork } from '@material-ui/icons';
import I18n from '@kevinwang0316/i18n';

import DashboardNumCard from './DashboardNumCard';
import { fetchInstanceCount as fetchInstanceCountAction } from '../../store/InstanceCount/actions';
import { InstanceCount as InstanceCountType } from '../../store/InstanceCount/types';
import { AppState } from '../../store/ConfigureStore';

interface Props {
  instanceCount: InstanceCountType;
  fetchInstanceCount: () => void;
}

const useStyles = makeStyles({
  cardIcon: {
    fontSize: 36,
    color: orange[500],
  },
});
let isFetching = false;

export const InstanceCountCard = (
  { instanceCount = null, fetchInstanceCount }: Props,
): ReactElement => {
  const classes = useStyles({});
  useEffect(() => {
    if (!instanceCount && !isFetching) {
      fetchInstanceCount();
      isFetching = true;
    }
  });

  return <DashboardNumCard displayNum={instanceCount} title={I18n.get('cardTitleWorkflowInstances')} icon={<GroupWork className={classes.cardIcon} />} />;
};

const mapStateToProps = (state: AppState) => ({ instanceCount: state.instanceCount });
const mapDispatchToProps = { fetchInstanceCount: fetchInstanceCountAction };
export default connect(mapStateToProps, mapDispatchToProps)(memo(InstanceCountCard));
