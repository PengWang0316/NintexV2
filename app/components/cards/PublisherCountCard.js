import React, { useEffect } from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import { connect } from 'react-redux';
import { lightGreen } from '@material-ui/core/colors';
import { SupervisedUserCircle } from '@material-ui/icons';
import I18n from '@kevinwang0316/i18n';

import DashboardNumCard from './DashboardNumCard';
import { fetchPublisherCount as fetchPublisherCountAction } from '../../actions/ActionActions';

const styles = {
  cardIcon: {
    fontSize: 36,
    color: lightGreen[500],
  },
};

export const PublisherCountCard = ({ classes, publisherCount, fetchPublisherCount }) => {
  useEffect(() => {
    if (!publisherCount) fetchPublisherCount();
  });

  return <DashboardNumCard displayNum={publisherCount} title={I18n.get('cardTitleWorkflowPublisher')} icon={<SupervisedUserCircle className={classes.cardIcon} />} />;
};

PublisherCountCard.propTypes = {
  classes: PropTypes.objectOf(PropTypes.string).isRequired,
  publisherCount: PropTypes.number,
  fetchPublisherCount: PropTypes.func.isRequired,
};
PublisherCountCard.defaultProps = {
  publisherCount: null,
};
const mapStateToProps = state => ({ publisherCount: state.publisherCount });
const mapDispatchToProps = { fetchPublisherCount: fetchPublisherCountAction };
export default connect(mapStateToProps, mapDispatchToProps)(withStyles(styles)(PublisherCountCard));
