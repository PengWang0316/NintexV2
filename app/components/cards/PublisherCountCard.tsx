import React, { useEffect, memo, ReactElement } from 'react';
import { makeStyles } from '@material-ui/core/styles';
import { connect } from 'react-redux';
import { lightGreen } from '@material-ui/core/colors';
import { SupervisedUserCircle } from '@material-ui/icons';
import I18n from '@kevinwang0316/i18n';

import DashboardNumCard from './DashboardNumCard';
import { fetchPublisherCount as fetchPublisherCountAction } from '../../store/PublisherCount/actions';
import { PublisherCount as PublisherCountType } from '../../store/PublisherCount/types';
import { AppState } from '../../store/ConfigureStore';

interface Props {
  publisherCount: PublisherCountType;
  fetchPublisherCount: () => void;
}

const useStyles = makeStyles({
  cardIcon: {
    fontSize: 36,
    color: lightGreen[500],
  },
});
let isFetching = false;

export const PublisherCountCard = (
  { publisherCount = null, fetchPublisherCount }: Props,
): ReactElement => {
  const classes = useStyles({});
  useEffect(() => {
    if (!publisherCount && !isFetching) {
      fetchPublisherCount();
      isFetching = true;
    }
  });

  return <DashboardNumCard displayNum={publisherCount} title={I18n.get('cardTitleWorkflowPublisher')} icon={<SupervisedUserCircle className={classes.cardIcon} />} />;
};

const mapStateToProps = (state: AppState) => ({ publisherCount: state.publisherCount });
const mapDispatchToProps = { fetchPublisherCount: fetchPublisherCountAction };
export default connect(mapStateToProps, mapDispatchToProps)((memo)(PublisherCountCard));
