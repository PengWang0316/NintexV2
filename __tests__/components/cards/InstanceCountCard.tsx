import React from 'react';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';

import { InstanceCountCard } from '../../../app/components/cards/InstanceCountCard';
import JssProviderWrapper from '../../libs/JssProviderWrapper';

describe('InstanceCountCard Component', () => {
  const defaultProps = {
    instanceCount: null,
    fetchInstanceCount: jest.fn(),
  };

  beforeEach(() => jest.clearAllMocks());

  test('useEffect call fetchInstanceCount', () => {
    render(<InstanceCountCard {...{ ...defaultProps }} />);
    expect(defaultProps.fetchInstanceCount).toHaveBeenCalledTimes(1);
  });

  test('Snapshot with null instanceCount', () => expect(renderer.create(JssProviderWrapper(<InstanceCountCard {...{ ...defaultProps }} />)).toJSON()).toMatchSnapshot());
  test('Snapshot with the instanceCount', () => expect(renderer.create(JssProviderWrapper(<InstanceCountCard {...{ ...defaultProps, instanceCount: 10 }} />)).toJSON()).toMatchSnapshot());
});
