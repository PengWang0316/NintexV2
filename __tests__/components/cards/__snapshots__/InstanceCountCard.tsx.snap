// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InstanceCountCard Component Snapshot with null instanceCount 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation1 MuiCard-root makeStyles-card-47 MuiPaper-rounded"
>
  <div
    className="MuiCardContent-root"
  >
    <div
      className="makeStyles-contentDiv-49"
    >
      <div>
        <div
          className="MuiBox-root MuiBox-root-80"
        >
          Workflow Instances
        </div>
        <div
          className="MuiBox-root MuiBox-root-81"
        >
          --
        </div>
      </div>
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root makeStyles-cardIcon-46"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zM8 17.5c-1.38 0-2.5-1.12-2.5-2.5s1.12-2.5 2.5-2.5 2.5 1.12 2.5 2.5-1.12 2.5-2.5 2.5zM9.5 8c0-1.38 1.12-2.5 2.5-2.5s2.5 1.12 2.5 2.5-1.12 2.5-2.5 2.5S9.5 9.38 9.5 8zm6.5 9.5c-1.38 0-2.5-1.12-2.5-2.5s1.12-2.5 2.5-2.5 2.5 1.12 2.5 2.5-1.12 2.5-2.5 2.5z"
        />
      </svg>
    </div>
  </div>
</div>
`;

exports[`InstanceCountCard Component Snapshot with the instanceCount 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation1 MuiCard-root makeStyles-card-47 MuiPaper-rounded"
>
  <div
    className="MuiCardContent-root"
  >
    <div
      className="makeStyles-contentDiv-49"
    >
      <div>
        <div
          className="MuiBox-root MuiBox-root-91"
        >
          Workflow Instances
        </div>
        <div
          className="MuiBox-root MuiBox-root-92"
        >
          10
        </div>
      </div>
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root makeStyles-cardIcon-46"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zM8 17.5c-1.38 0-2.5-1.12-2.5-2.5s1.12-2.5 2.5-2.5 2.5 1.12 2.5 2.5-1.12 2.5-2.5 2.5zM9.5 8c0-1.38 1.12-2.5 2.5-2.5s2.5 1.12 2.5 2.5-1.12 2.5-2.5 2.5S9.5 9.38 9.5 8zm6.5 9.5c-1.38 0-2.5-1.12-2.5-2.5s1.12-2.5 2.5-2.5 2.5 1.12 2.5 2.5-1.12 2.5-2.5 2.5z"
        />
      </svg>
    </div>
  </div>
</div>
`;
